spring-security/src/main/java/com/sap/cloud/security/spring/token/authentication/ReactiveHybridJwtDecoder.java

```diff
@@ -59,10 +59,15 @@ public Mono<Jwt> decode(String encodedToken) throws JwtException {
     }
 
     static Mono<Jwt> parseJwt(Token token) {
-        Instant issuedAt = token.hasClaim(TokenClaims.XSUAA.ISSUED_AT)
-                ? Instant.ofEpochSecond(Long.parseLong(token.getClaimAsString(TokenClaims.XSUAA.ISSUED_AT)))
-                : null;
-        return Mono.just(new Jwt(token.getTokenValue(), issuedAt,
-                token.getExpiration(), token.getHeaders(), token.getClaims()));
+        try{
+            Instant issuedAt = token.hasClaim(TokenClaims.XSUAA.ISSUED_AT)
+                    ? Instant.ofEpochSecond(Long.parseLong(token.getClaimAsString(TokenClaims.XSUAA.ISSUED_AT)))
+                    : null;
+            return Mono.just(new Jwt(token.getTokenValue(), issuedAt,
+                    token.getExpiration(), token.getHeaders(), token.getClaims()));
+        }
+        catch (NumberFormatException e){
+            throw new RuntimeException("Error parsing JWT: " + e.getMessage(), e);
```

Comment:
I would prefer here to use BadJwtException instead of the generic runtime exception